source/equipment/equipment_api.cpp

```diff
@@ -0,0 +1,99 @@
+/* Copyright © 2001-2019, Canal TP and/or its affiliates. All rights reserved.
+
+This file is part of Navitia,
+    the software to build cool stuff with public transport.
+
+Hope you'll enjoy and contribute to this project,
+    powered by Canal TP (www.canaltp.fr).
+Help us simplify mobility and open public transport:
+    a non ending quest to the responsive locomotion way of traveling!
+
+LICENCE: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#include "equipment_api.h"
+#include "utils/paginate.h"
+
+#include "boost/range/algorithm.hpp"
+
+#include <map>
+#include <vector>
+#include <string>
+
+namespace navitia {
+namespace equipment {
+
+namespace {
+    const std::string build_ptref_line_filter(const std::string& line_uri, const std::string& filter) {
+        std::string line_filter = "line.uri=" + line_uri;
+        if(filter.size()) {
+            line_filter += " and (" + filter + ")";
+        }
+        return line_filter;
+    }
+
+    void fill_equipment_to_pb(const equipment::StopAreasPerLine& sas_per_line,
+                              PbCreator& pb_creator, int depth) {
+        for(const auto& line : sas_per_line) {
+            auto report = pb_creator.add_equipment_reports();
+            pb_creator.fill(line.first, report->mutable_line(), depth);
+            for(const auto& sa : line.second) {
+                auto sa_equip = report->add_stop_area_equipments();
+                pb_creator.fill(sa, sa_equip->mutable_stop_area(), depth);
```

Comment:
I think that we should fix a min depth, as we need stop_points and their codes to be able to fill it later in jormun, no?
When requesting `v1/stop_areas?depth=0` we get no SP, so that could be a problem (not sure).

Reply:
as we mentioned, we'll not be using the stop-points to get the codes from them. I've actually forgotten to fill the protobuf properly :confounded: